Updating tokio-util dependency in minor version broke semver

#270 updated the externally visible tokio-util dependency across a semver-incompatible boundary (0.6 to 0.7), and the affected crates were subsequently released without incrementing the major version. This breaks crates depending on them, e.g.:

```
$ cargo new foo
$ cd foo
$ echo -e 'hyper-websocket-lite = "0.5.0"\ntokio = {version = "1", features = ["rt"]}' >> Cargo.toml
$ cargo build
[...]
error[E0599]: no method named `framed` found for struct `MessageCodec` in the current scope
   --> /home/xiretza/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-websocket-lite-0.5.0/src/lib.rs:41:53
    |
41  |                 let client = MessageCodec::server().framed(upgraded);
    |                                                     ^^^^^^ method not found in `MessageCodec`
    |
   ::: /home/xiretza/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-util-0.7.3/src/codec/decoder.rs:178:8
    |
178 |     fn framed<T: AsyncRead + AsyncWrite + Sized>(self, io: T) -> Framed<T, Self>
    |        ------ the method is available for `MessageCodec` here
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
6   | use tokio_util::codec::decoder::Decoder;
    |
```

This is because `hyper-websocket-lite 0.5.0` depends both on `tokio-util 0.6` and `websocket-codec`, but the latest compatible `websocket-codec` version (`0.5.2`) now pulls in `tokio-util 0.7` (due to the `>= 0.6` constraint). Thus, `hyper-websocket-lite`'s `tokio_util::codec::decoder::Decoder` (from `tokio-util 0.6`) is different from `websocket-codec`'s `tokio_util::codec::decoder::Decoder` (from `tokio-util 0.7`).

The proper solution here is to yank all affected versions (i.e. `websocket-codec` and `websocket-lite` versions `0.5.1` and `0.5.2`), republish the crates under a new major version, and avoid `>=` dependencies like the plague - they undermine the whole purpose of semver, since there's no way for you to actually guarantee that you will remain compatible with any breaking changes of the dependency.

--- a/fuzz/Cargo.toml
+++ b/fuzz/Cargo.toml
@@ -11,7 +11,7 @@ cargo-fuzz = true
 [dependencies]
 bytes = "1"
 libfuzzer-sys = "0.4"
-tokio-util = { version=">=0.6", features=["codec"] }
+tokio-util = { version="0.7", features=["codec"] }
 websocket-codec = { path = "../websocket-codec" }
 
 [[bin]]

--- a/hyper-websocket-lite/Cargo.toml
+++ b/hyper-websocket-lite/Cargo.toml
@@ -12,7 +12,7 @@ futures = "0.3"
 hyper = { version = "0.14", features=["http1", "server"] }
 websocket-codec = { version = "0.5", path = "../websocket-codec" }
 tokio = { version = "1", features=["macros"] }
-tokio-util = { version = ">=0.6", features=["codec"] }
+tokio-util = { version = "0.7", features=["codec"] }
 
 [dev-dependencies]
 hyper = { version = "0.14", features=["tcp"] }

--- a/websocket-codec/Cargo.toml
+++ b/websocket-codec/Cargo.toml
@@ -15,7 +15,7 @@ bytes = "1"
 httparse = "1"
 rand = "0.8"
 sha1 = "0.6"
-tokio-util = { version=">=0.6", features=["codec"] }
+tokio-util = { version="0.7", features=["codec"] }
 
 [dev-dependencies]
 assert-allocations = { path="../assert-allocations" }

--- a/websocket-lite/Cargo.toml
+++ b/websocket-lite/Cargo.toml
@@ -15,7 +15,7 @@ futures = "0.3"
 native-tls = { version = "0.2", optional = true }
 openssl = { version = "0.10", optional = true }
 rand = "0.8"
-tokio-util = { version = ">=0.6", features = ["codec"] }
+tokio-util = { version = "0.7", features = ["codec"] }
 tokio-openssl = { version = "0.6", optional = true }
 tokio-native-tls = { version = "0.3", optional = true }
 tokio = { version = "1", features = ["net", "io-util"] }

